### PR TITLE
Refactor autopilot package

### PR DESCRIPTION
**Pull request by hidmic**
# Description

Depends on #133. This patch:

* Drops the independent thread in favor of a `spin()` method
* Replaces boolean flags with type-safe "bitmask".
* Cleans up code, pruning redundant execution paths.
* Normalizes code style 

This should get rid of the issues reported in #133.

## Type of change

Please mark options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Unit tests are passing
- [x] Linter tests are passing

# How To Test

Run all tests on the vehicle:

```sh
catkin_make -j1 run_tests
```

All should pass. Doing some manual QA would also be advisable.

### CODE REVIEW

File: autopilot/src/autopilot.cpp [hunk 1]
```diff
@@ -36,549 +36,566 @@
...
-    desiredSpeed = 0;
-    desiredRoll = 0;
-    desiredYaw = 0;
-    desiredPitch = 0;
-    desiredDepth = 0;
-    desiredRudder = 0;
-    fixedRudder = true;
-    depthControl = false;
-    altitudeControl = false;
+    if (auto_pilot_in_control_) {
```
Comment (patapufate) [inline autopilot/src/autopilot.cpp:276]: nit: should be on a line by itself

File: autopilot/src/autopilot.cpp [hunk 2]
```diff
@@ -36,549 +36,566 @@
...
+      // clear the following so we do not take off after manual control is released.
+      active_setpoints_ = {};
+      auto_pilot_in_control_ = false;
+    }
   }
   else
   {
-    // ROS_INFO("Jaus says its not in control.");
-    autoPilotInControl = true;
+    if (!auto_pilot_in_control_) {
```
Comment (patapufate) [inline autopilot/src/autopilot.cpp:285]: nit: should be on a line by itself

**Review by patapufate — APPROVED** [review]
LGTM!

File: autopilot/src/autopilot.cpp (hunk 1 above)
Comment (hidmic) [reply autopilot/src/autopilot.cpp]: Indeed. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/134/commits/b72ee3a4d2a1caa026f360e9708ca5cb5537bb7e.

File: autopilot/src/autopilot.cpp (hunk 2 above)
Comment (hidmic) [reply autopilot/src/autopilot.cpp]: Indeed. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/134/commits/b72ee3a4d2a1caa026f360e9708ca5cb5537bb7e.

**Comment by hidmic** [comment]
Rebased to drop #133 commits.

**Comment by hidmic** [comment]
Alright, tests pass now. Manual QA, however, is showing autopilot issues both in `master` and this branch. We have not identified the root cause yet.

Thus, we'll merge this PR as-is and reassess the situation in `master`.

**Pull request merged**